crates/actors/src/mining.rs

```diff
@@ -91,15 +90,16 @@ impl PartitionMiningActor {
             } else {
                 next_ranges_step
             };
-            // check if we need to reconstruct steps, that is inverval start..=step-1 is not empty
-            if start <= step - 1 {
+            // check if we need to reconstruct steps, that is interval start..=step-1 is not empty
+            if start < step {
                 debug!("Getting stored steps from ({}..={})", start, step - 1);
                 let vdf_steps = self.steps_guard.read();
                 let steps = vdf_steps.get_steps(ii(start, step - 1))?; // -1 because last step is calculated in next get_recall_range call, with its corresponding argument seed
                 self.ranges.reconstruct(&steps, partition_hash);
             };
-            Ok(self.ranges.get_recall_range(step, seed, partition_hash) as u64) // calculates step range
         }
+
+        Ok(self.ranges.get_recall_range(step, seed, partition_hash) as u64) // calculates step range
```

Comment:
```suggestion
        self.ranges.get_recall_range(step, seed, partition_hash).try_into().wrap_err("recall range larger than u64") // calculates step range
```

getting rid of `as` conversions is good practice, we don't want any unintended behavior stemming from that.